rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java

```diff
@@ -808,6 +758,90 @@ Collection<MavenRepository> distinctNormalizedRepositories(
+        // Always prefer to use https, fallback to http only if https isn't available
+        // URLs are case-sensitive after the domain name, so it can be incorrect to lowerCase() a whole URL
+        // This regex accepts any capitalization of the letters in "http"
+        String originalUrl = repository.getUri();
+        String httpsUri = originalUrl.toLowerCase().startsWith("http:") ?
+                repository.getUri().replaceFirst("[hH][tT][tT][pP]://", "https://") :
+                repository.getUri();
+        if (!httpsUri.endsWith("/")) {
+            httpsUri += "/";
+        }
+
+        HttpSender.Request.Builder request = httpSender.options(httpsUri);
+        if (repository.getTimeout() != null) {
+            request = request.withConnectTimeout(repository.getTimeout())
+                    .withReadTimeout(repository.getTimeout());
+        }
+
+        ReachabilityResult reachability = reachable(repository, applyAuthenticationToRequest(repository, request));
+        if (reachability.isSuccess()) {
+            return repository.withUri(httpsUri);
+        }
+        reachability = reachable(repository, applyAuthenticationToRequest(repository, request.withMethod(HttpSender.Method.HEAD).url(httpsUri)));
+        if (reachability.isReachable()) {
+            return repository.withUri(httpsUri);
+        }
+        if (!originalUrl.equals(httpsUri)) {
+            reachability = reachable(repository, applyAuthenticationToRequest(repository, request.withMethod(HttpSender.Method.OPTIONS).url(originalUrl)));
+            if (reachability.isSuccess()) {
+                return repository.withUri(originalUrl);
+            }
+            reachability = reachable(repository, applyAuthenticationToRequest(repository, request.withMethod(HttpSender.Method.HEAD).url(originalUrl)));
+            if (reachability.isReachable()) {
+                return repository.withUri(originalUrl);
+            }
+        }
+        // Won't be null if server is unreachable
+        throw Objects.requireNonNull(reachability.throwable);
+    }
+
+    @Value
+    private static class ReachabilityResult {
+        Reachability reachability;
+        @Nullable
+        Throwable throwable;
+
+        private static ReachabilityResult success() {
+            return new ReachabilityResult(Reachability.SUCCESS, null);
+        }
+
+        public boolean isReachable() {
+            return reachability == Reachability.SUCCESS || reachability == Reachability.ERROR;
+        }
+
+        public boolean isSuccess() {
+            return reachability == Reachability.SUCCESS;
+        }
+    }
+
+    enum Reachability {
+        SUCCESS,
+        ERROR,
+        UNREACHABLE;
+    }
+
+    private ReachabilityResult reachable(MavenRepository repository, HttpSender.Request.Builder requestBuilder) {
+        HttpSender.Request.Builder request = applyAuthenticationToRequest(repository, requestBuilder);
+        try {
+            sendRequest(request.build());
```

Comment:
Calls into `reachable` also seem to call `applyAuthenticationToRequest` themselves before passing in their `requestBuilder`. Should we leave that up to the callers given the OPTIONS without auth headers above?
```suggestion
        try {
            sendRequest(requestBuilder.build());
```

Reply:
good point. Updated